cmd/podman/search.go

```diff
@@ -0,0 +1,298 @@
+			limit = opts.limit
+		}
+
+		for i := 0; i < limit; i++ {
+			if len(opts.filter) > 0 {
+				// Check whether query matches filters
+				if !(matchesAutomatedFilter(filter, val[i]) && matchesOfficialFilter(filter, val[i]) && matchesStarFilter(filter, val[i])) {
+					continue
+				}
+			}
+			official := ""
+			if val[i].IsOfficial {
+				official = "[OK]"
+			}
+			automated := ""
+			if val[i].IsAutomated {
+				automated = "[OK]"
+			}
+			description := val[i].Description
+			if len(description) > 44 && !opts.noTrunc {
+				description = description[:descriptionTruncLength] + "..."
+			}
+			name := index + "/" + val[i].Name
+			if index == "docker.io" && !strings.Contains(val[i].Name, "/") {
+				name = index + "/library/" + val[i].Name
+			}
+			params := searchParams{
+				Index:       index,
+				Name:        name,
+				Description: description,
+				Official:    official,
+				Automated:   automated,
+				Stars:       val[i].StarCount,
+			}
+			paramsArr = append(paramsArr, params)
+		}
+	}
+	return paramsArr, nil
+}
+
+func generateSearchOutput(term string, registries []string, opts searchOpts, filter searchFilterParams) error {
+	searchOutput, err := getSearchOutput(term, registries, opts, filter)
+	if err != nil {
+		return err
+	}
+	if len(searchOutput) == 0 {
+		return nil
+	}
+	out := formats.StdoutTemplateArray{Output: searchToGeneric(searchOutput), Template: opts.format, Fields: searchOutput[0].headerMap()}
+	return formats.Writer(out).Out()
+}
+
+func parseSearchFilter(opts *searchOpts) (*searchFilterParams, error) {
+	filterParams := &searchFilterParams{}
+	for _, filter := range opts.filter {
+		arr := strings.Split(filter, "=")
+		switch arr[0] {
+		case "stars":
+			if len(arr) < 2 {
+				return nil, errors.Errorf("invalid `stars` filter %q, should be stars=<value>", filter)
+			}
+			stars, err := strconv.Atoi(arr[1])
+			if err != nil {
+				return nil, errors.Wrapf(err, "incorrect value type for stars filter")
+			}
+			filterParams.stars = stars
+			break
+		case "is-automated":
+			if len(arr) == 2 {
+				if arr[1] == "true" {
+					filterParams.isAutomated = true
+				} else {
+					filterParams.isAutomated = false
+				}
+			} else {
+				filterParams.isAutomated = true
+			}
+			filterParams.isAutomatedSet = true
+			break
+		case "is-official":
+			if len(arr) == 2 {
+				if arr[1] == "true" {
+					filterParams.isOfficial = true
+				} else {
+					filterParams.isOfficial = false
+				}
+			} else {
+				filterParams.isOfficial = true
+			}
+			filterParams.isOfficialSet = true
+			break
+		default:
+			return nil, errors.Errorf("invalid filter type %q", filter)
+		}
+	}
+	return filterParams, nil
+}
+
+func matchesStarFilter(filter searchFilterParams, result docker.Results) bool {
+	return result.StarCount >= filter.stars
+}
+
+func matchesAutomatedFilter(filter searchFilterParams, result docker.Results) bool {
+	if filter.isAutomatedSet {
+		return result.IsAutomated == filter.isAutomated
```

Comment:
Should not need isAutomatedSet and return result.IsAutomated

Reply:
I don't think I am getting what you mean. If I don't have the `isAutomatedSet` field, and just compare the `isAutomated` field of the result to the filter, I will get only the results that have this set to false or true based on the value of `filter.isAutomated`. But I want all the results regardless of the `isAutomated` value when the `is-automated` filter is not set.

Reply:
if isAutomated is defaulted to true. Then only changed if the user specifies to remove it with automated == false.

Reply:
You are always printing it unless the user tells you not to, so default to true and set it to false when the user tells, then you don't need the rest of the code.

Reply:
If that is the case, then when the user only wants images with isAutomated true, they will get all the images regardless of the images isAutomated value. It will work for the false case, but not the true case.
These are the cases:
1) --filter=isAutomated=true -> Want all images with isAutomated set to true
2) --filter =isAutomated=false -> Want all images with isAutomated set to false
3) no filter set -> Want all images regardless of the image's isAutomated value

Reply:
Ok, I still think we could get by with one variable.  
Automated String
Set it to "true", "false" or "" and that gets your three states.

Reply:
I would go with *bool - that gives us true, false, and nil

Reply:
Right, forgot we could use pointers. Fixed